### PR TITLE
Code style

### PR DESCRIPTION
**Pull request by robsdedude**
Let's make Guido van Bob Rossum happy.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        additional_dependencies:
+          - flake8-bugbear
+          - flake8-builtins
+          - flake8-docstrings
+          - flake8-quotes
+          - pep8-naming
+        args:
+          - "--inline-quotes=double"
+          - "--ignore=A003,E241,W503"
+          - "--extend-ignore=D100,D101,D102,D103,D104,D105,D106,D107"
```
Comment (robsdedude) [inline .pre-commit-config.yaml]:
```suggestion
          - "--ignore=A003,D100,D101,D102,D103,D104,D105,D106,D107,E241,W503"
```

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# based_on_style = "pep8"
+# COALESCE_BRACKETS = true
+# ALIGN_CLOSING_BRACKET_WITH_VISUAL_INDENT = true
+# DEDENT_CLOSING_BRACKETS = false
+# INDENT_DICTIONARY_VALUE = true
+# SPLIT_BEFORE_ARITHMETIC_OPERATOR = true
+# SPLIT_BEFORE_DOT = true
+# SPLIT_BEFORE_EXPRESSION_AFTER_OPENING_PAREN = true
+# SPLIT_BEFORE_FIRST_ARGUMENT = false
+# SPLIT_COMPLEX_COMPREHENSION = true
```
Comment (robsdedude) [inline pyproject.toml]:
```suggestion
```

File: requirements.txt [hunk 3]
```diff
@@ -1,3 +1,5 @@
+ifaddr
+isort
```
Comment (robsdedude) [inline requirements.txt]:
```suggestion
```

**Review by bigmontz — APPROVED** [review]
👔 

Just one question: does the pre-commit only checks in the staged files?

**Comment by robsdedude** [comment]
> Just one question: does the pre-commit only checks in the staged files?

I've tried it out real quick and, yes, it looks like it.

File: boltstub/simple_jolt/transformers.py [hunk 4]
```diff
@@ -228,7 +228,7 @@ def _decode_full(value, decode_cb):
                 or (isinstance(value, list)
                     and all(isinstance(b, int) and 0 <= b <= 255
                             for b in value))):
-            raise JOLTValueError('Expected str or list of integers (0-255) '
+            raise JOLTValueError("Expected str or list of integers (0-255) "
```
Comment (AndyHeap-NeoTech) [inline boltstub/simple_jolt/transformers.py:231]: does mixing single quote and double quote strings work ok? See following line where it uses single quotes.

**Review by AndyHeap-NeoTech — APPROVED** [review]

File: boltstub/simple_jolt/transformers.py (hunk 4 above)
Comment (robsdedude) [reply boltstub/simple_jolt/transformers.py:231]: Yes is does, no I don't particularly like it. I prefer: double quotes, unless you have `"` in the string + I prefer to have a consistent quotation symbol for wrapped strings like this one. The analyzer currently does not support the second aspect (https://github.com/zheller/flake8-quotes/issues/82). I could work around this by adding `# noqa: Q000` at the end of each conflicting line, but that makes the code base uglier than mixing sting styles, imo.

**Pull request merged**